src/Microsoft.OData.Core/ODataMessageReaderSettings.cs

```diff
@@ -235,6 +236,12 @@ public ODataMessageQuotas MessageQuotas
         /// </summary>
         internal bool ThrowOnUndeclaredPropertyForNonOpenType { get; private set; }
 
+        /// <summary>
+        /// Require Content-Id header in changesets
+        /// If turned off allows to read OData 2.0 requests without Content-Id header present.

```

Comment:
What do you mean about "OData 2.0" ? It's a OData spec version number?

Reply:
To be precise this small addition not to support 4.0 metadata, just for reading requests